Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -167,25 +167,54 @@ public protocol PackageContainerProvider {
     func getContainer(for identifier: Container.Identifier, completion: @escaping (Result<Container, AnyError>) -> Void)
 }
 
+/// The requirement of this constraint.
+public enum Requirement<T: PackageContainerIdentifier>: Equatable {
```

Comment:
Did this have to be lifted to a top-level type? I liked it inside Constraint TBH.

If we do have to lift it it should be PackageContainerRequirement probably.